tokens/usecase/tokens_usecase_test.go

```diff
@@ -539,5 +542,98 @@ func (s *TokensUseCaseTestSuite) TestGetMinPoolLiquidityCap() {
 			s.Require().Equal(tt.expectedMinPoolLiquidityCap, actualMinPoolLiquidityCap)
 		})
 	}
+}
+
+// Test to validate valid chain denom works as expected.
+func (s *TokensUseCaseTestSuite) TestIsValidChainDenom() {
+	testcases := []struct {
+		name           string
+		chainDenom     string
+		tokens         map[string]domain.Token
+		expectedResult bool
+	}{
+		{
+			name:           "Valid chain denom",
+			chainDenom:     "validDenom",
+			tokens:         map[string]domain.Token{"validDenom": {IsUnlisted: false}},
+			expectedResult: true,
+		},
+		{
+			name:           "Invalid chain denom - not found",
+			chainDenom:     "invalidDenom",
+			tokens:         map[string]domain.Token{"validDenom": {IsUnlisted: false}},
+			expectedResult: false,
+		},
+		{
+			name:           "Invalid chain denom - unlisted",
+			chainDenom:     "unlistedDenom",
+			tokens:         map[string]domain.Token{"unlistedDenom": {IsUnlisted: true}},
+			expectedResult: false,
+		},
+	}
+
+	for _, tt := range testcases {
+		s.Run(tt.name, func() {
+			usecase := tokensusecase.NewTokensUsecase(tt.tokens, 0, nil, nil)
+			result := usecase.IsValidChainDenom(tt.chainDenom)
+			s.Require().Equal(tt.expectedResult, result)
+		})
+	}
+}
+
+// mocktokenLoader is a mock implementation of the tokensusecase.TokenLoader interface.
+type mockTokenLoader struct {
+	callback func()
+}
+
+// FetchAndUpdateTokens implements the tokensusecase.TokenLoader interface.
+func (m *mockTokenLoader) FetchAndUpdateTokens(loadTokens tokensusecase.LoadTokensFunc) error {
+	m.callback()
+	return nil
+}
+
+// TestUpdateAssetsAtHeightIntervalAsync tests the async update of assets at height interval.
+func (s *TokensUseCaseTestSuite) TestUpdateAssetsAtHeightIntervalAsync() {
+	logger, err := log.NewLogger(false, "", "")
+	s.Require().NoError(err)
+
+	updateAssetsHeightInterval, gen, iterations := rand.Intn(100), rand.Intn(8932122), rand.Intn(1000)
```

Comment:
Ditto for using deterministic seeds as needed. Generally, I recommend opting for table-driven test cases , where possible, making the async version a simple propagator/wrapper to a heavily unit-tested synchronous method